### PR TITLE
[idlharness] Support testing of [LegacyWindowAlias] extended attribute

### PR DESCRIPTION
**Pull request by zcorpan**
Follows https://github.com/heycam/webidl/pull/364

### CODE REVIEW

**Comment by zcorpan** [comment]
The new test passes in Chromium with this ad-hoc IDL file:
```
[LegacyWindowAlias=webkitURL]
interface URL {};
```

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233229785)

# Firefox (nightly) #
Testing web-platform-tests at revision 5454b89ffbf5b110dd206228508756e1e6eb06bb
Ignoring 1 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233229786)

# Sauce (safari) #
Testing web-platform-tests at revision 5454b89ffbf5b110dd206228508756e1e6eb06bb
Ignoring 1 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233229794)

# Chrome (unstable) #
Testing web-platform-tests at revision 5454b89ffbf5b110dd206228508756e1e6eb06bb
Ignoring 1 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/233229795)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 5454b89ffbf5b110dd206228508756e1e6eb06bb
Ignoring 1 changed files:
No files changed

File: resources/idlharness.js [hunk 1]
```diff
@@ -972,6 +972,51 @@ IdlInterface.prototype.test_self = function()
...
+        {
+            var aliasAttrs = this.extAttrs.filter(function(o) { return o.name === "LegacyWindowAlias"; });
+            if (aliasAttrs.length > 1) {
+                throw "Invalid IDL: multiple LegacyWindowAlias extended attributes on " + this.name;
+            }
+            if (this.is_callback()) {
+                throw "Invalid IDL: LegacyWindowAlias extended attribute on non-interface " + this.name;
+            }
+            if (this.exposureSet.indexOf("Window") === -1) {
+                throw "Invalid IDL: LegacyWindowAlias extended attribute on " + this.name + " but not exposed in Window";
```
Comment (tobie) [inline resources/idlharness.js]: s/but not/which is not/ ?

File: resources/idlharness.js [hunk 2]
```diff
@@ -972,6 +972,51 @@ IdlInterface.prototype.test_self = function()
...
+            var aliases;
+            if (rhs.type === "identifier-list") {
+                aliases = rhs.value;
+            } else { // rhs.type === identifier
+                aliases = [ rhs.value ];
+            }
+
+            // OK now actually check the aliases...
+            for (var alias of aliases) {
+                assert_true(alias in self, alias + " should exist");
```
Comment (tobie) [inline resources/idlharness.js]: Isn't that going to fail in workers?

**Review by tobie — COMMENTED** [review]
Are these tests prevented from running in workers (per spec)?

**Comment by zcorpan** [comment]
@tobie PTAL

**Review by tobie — APPROVED** [review]
LGTM

**Pull request merged**